frontend-cert/react-projects/react-use-state/index.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Color Picker</title>
+    <link rel="stylesheet" href="styles.css" />
+    <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
+    <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+</head>
+
+<body>
+    <div id="root"></div>
+    <script type="text/babel">
+        const { useState } = React;
+
+        function MyApp() {
+            const [bgColor, setBgColor] = useState('#ffffff'); // Initial background color
+
+            const handleButtonClick = () => {
+                document.getElementById('color-input').click(); // Programmatically click the hidden color input
+            };
```

Comment:
this is an anti pattern in react. it would be better to use a ref here. I am not looking at the sheet but I don't believe refs have been taught yet.


Instead of the pick a color button, you can have some text like this


```html
   <div id="color-picker-container" style={{ backgroundColor: bgColor }}>
            <p style={{ margin: 0 }}>
              Choose a color using the color input below:
            </p>

            <input
              id="color-input"
              type="color"
              value={bgColor}
              onChange={handleColorChange}
            />
          </div>
```

Reply:
Now the prototype looks like this:

![image](https://github.com/user-attachments/assets/857f5392-1723-4b2e-a7a0-f1836fa9d075)